lib/bike_brigade_web/live/program_live/form_component.html.heex

```diff
@@ -9,149 +9,157 @@
-              phx-click={JS.push("remove_schedule", value: %{index: i}, target: @myself)}
-              type="button"
-            >
-              <span class="md:hidden mr-2 text-gray-700">Remove Schedule</span>
-              <Heroicons.x_mark solid class="w-6 h-6" />
-            </.button>
-          </div>
-          <hr class="mb-2 md:hidden"/>
+          <.inputs_for :let={s} field={c[:schedules]}>
+            <div class="flex flex-col space-x-0 space-y-2 md:flex-row md:items-center md:space-y-1 md:space-x-2">
+              <.input
+                type="select"
+                class="mt-8"
+                field={{s, :weekday}}
+                options={[
+                  Monday: :monday,
+                  Tuesday: :tuesday,
+                  Wednesday: :wednesday,
+                  Thursday: :thursday,
+                  Friday: :friday,
+                  Saturday: :saturday,
+                  Sunday: :sunday
+                ]}
+              />
+              <.input type="time" field={{s, :start_time}} />
+              <.input type="time" field={{s, :end_time}} />
+              <div class="justify-center mt-1 ml-auto">
+                <.button
+                  color={:clear}
+                  size={:xsmall}
+                  phx-click={
+                    JS.push("remove_schedule", value: %{index: s.index}, target: @myself)
+                  }
+                  type="button"
+                >
+                  <span class="mr-2 text-gray-700 md:hidden">Remove Schedule</span>
+                  <Heroicons.x_mark solid class="w-6 h-6" />
+                </.button>
+              </div>
+              <hr class="mb-2 md:hidden" />
+            </div>
+          </.inputs_for>
         </div>
-      </div>
 
-      <div :if={@action == :edit} class="my-2">
-        <.header small>
-          Items
-          <:subtitle>
-            What we deliver for this program
-          </:subtitle>
-          <:actions>
-            <.button size={:xsmall} phx-click={JS.push("add_item", target: @myself)}>
-              New Item
-            </.button>
-          </:actions>
-        </.header>
+        <div :if={@action == :edit} class="my-2">
+          <.header small>
+            Items
+            <:subtitle>
+              What we deliver for this program
+            </:subtitle>
+            <:actions>
+              <.button size={:xsmall} phx-click={JS.push("add_item", target: @myself)}>
+                New Item
+              </.button>
+            </:actions>
+          </.header>
 
-        <.table id="items" rows={inputs_for(f, :items) |> Enum.with_index()}>
-          <:col :let={{item_form, _i}} label="Name">
-            <.input type="hidden" field={{item_form, :id}} />
-            <.input type="hidden" field={{item_form, :program_id}} />
-            <.input type="text" field={{item_form, :name}} />
-          </:col>
+          <.table id="items" rows={f.impl.to_form(f.source, f, :items, [])}>
```

Comment:
Ooof I am not happy with this, but can't think of a better way

Reply:
Curious, where in the docs did you find the `.impl.to_form` bit? I was looking around for how to solve this at one point and didn't find anything.

Reply:
Thanks for calling this out!

I found the `impl.to_form` by reading the code for `inputs_for`. But this wasn't the right way to do it - calling an impl directly is probably a bad idea, instead I switched to calling via the protocol `Phoenix.HTML.FormData.to_form/4`